lib/PrivateBin.php

```diff
@@ -369,12 +366,15 @@ private function _read($dataid)
         try {
             $paste = $this->_model->getPaste($dataid);
             if ($paste->exists()) {
-                $data              = $paste->get();
-                $this->_doesExpire = property_exists($data, 'meta') && property_exists($data->meta, 'expire_date');
-                if (property_exists($data->meta, 'salt')) {
-                    unset($data->meta->salt);
+                // reading paste is only possible via JSON call
+                if ($this->_request->isJsonApiCall()) {
```

Comment:
Hmm, if this check fails, it does not result in an error, does it? (I know, my version before also did not, but mhh…)

Maybe to clarify the code we could also do early returns here,  i.e. don't nest if statements, but return them. But this is only a minor idea, and seeing that there is some error logic below, it does not seem possible so easy.